direct/src/dist/pip_wrapper.py

```diff
@@ -0,0 +1,86 @@
+from __future__ import print_function
+
+import platform
+import pip
+import sys
+import os
+
+target_platform = None
+sys_platform = None
+platform_machine = None
+platform_system = None
+os_name = None
+
+for (index, arg) in enumerate(sys.argv):
+    if arg == '--platform':
+        target_platform = sys.argv[index + 1]
+
+if target_platform is not None:
+    if target_platform.startswith("win"):
+        sys_platform = 'win32'
+        os_name = 'nt'
+    elif target_platform.startswith("manylinux") or target_platform.startswith("linux"):
+        sys_platform = 'linux'
+        os_name = 'posix'
+    elif target_platform.startswith('macosx'):
+        sys_platform = 'darwin'
+        os_name = 'posix'
+    else:
+        print("Unsupported target platform")
+    for machine in ("x86_64", "amd64", "i386", "i586", "i686", "ppc"):
+        if target_platform.endswith(machine):
+            platform_machine = machine
+            break
+    if platform_machine is None:
```

Comment:
This construct is fine and may want to be kept for readability, but this type of pattern is what for-else is for.

Reply:
I'm not a big fan of for-else but I can use it there if preferred